### PR TITLE
[10.0][MIG] barcodes_generator_product

### PR DESCRIPTION
**Pull request by lasley**
Migration of `barcodes_product_generator` to v10.

* Rename manifest
* Change openerp references to odoo
* Bump version

Depends:
~- [ ] #65~

Todos:
- [x] This should remove/deprecate the old module `product_barcode_generator` after confirming that is the intent of this module with @legalsylvain

### CODE REVIEW

File: barcodes_generator_abstract/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    'name': 'Generate Barcodes (Abstract)',
+    'summary': 'Generate Barcodes for Any Models',
+    'version': '10.0.1.0.0',
+    'category': 'Tools',
+    'author':
+        'GRAP,'
+        'La Louve,'
+        'Odoo Community Association (OCA)',
+    'website': 'http://www.odoo-community.org',
```
Comment (lasley) [inline barcodes_generator_abstract/__manifest__.py]: https

File: barcodes_generator_product/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    'name': 'Generate Barcodes for Products',
+    'summary': 'Generate Barcodes for Products (Templates and Variants)',
+    'version': '10.0.1.0.0',
+    'category': 'Tools',
+    'author':
+        'GRAP,'
+        'La Louve,'
+        'Odoo Community Association (OCA)',
+    'website': 'http://www.odoo-community.org',
```
Comment (lasley) [inline barcodes_generator_product/__manifest__.py]: https

**Comment by legalsylvain** [comment]
Hi, 

I'm agree to remove product_barcode_generator from the 10.0 repository. The current module covers product_barcode_generator feature and add more features, based on standard model.

@ all, as I'm the author of this new module, an external point of view could be interesting.

The point of view of the authors of the original module could be great too.

CC : @yelizariev, @damdam-s.

File: barcodes_generator_product/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    ],
+    'data': [
+        'views/view_product_product.xml',
+        'views/view_product_template.xml',
+    ],
+    'demo': [
+        'demo/res_users.xml',
+        'demo/barcode_rule.xml',
+        'demo/product.xml',
+        'demo/function.xml',
```
Comment (legalsylvain) [inline barcodes_generator_product/__manifest__.py:30]: I don't see xml files, and __init_ file. I think that your PR is not complete. Or did I missed something?

**Review by legalsylvain — CHANGES_REQUESTED** [review]

File: barcodes_generator_product/__manifest__.py (hunk 3 above)
Comment (lasley) [reply barcodes_generator_product/__manifest__.py:30]: Oops something wonky must have gone one with my rebase. Good to go now!

I'll check the other ones too, I typically do my rebases on after another so I bet I broke them too 😆

**Comment by lasley** [comment]
To all reviewers, the relevant commits are:
* 0229303 - barcodes_product_generator v10 migration
* e26ea46 - product_barcode_generator deprecation

**Review by legalsylvain — APPROVED** [review]
:+1: Thanks.
Change review, no test.

File: barcodes_generator_product/README.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================================================
+Generate Barcodes for Products (Templates and Variants)
+=======================================================
+
+This module expends Odoo functionality, allowing user to generate barcode
```
Comment (pedrobaeza) [inline barcodes_generator_product/README.rst]: s/expends/expands

File: barcodes_generator_product/README.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+depending on a given barcode rule for Products.
+
+For exemple, a typical pattern for products is  "20.....{NNNDD}" that means
```
Comment (pedrobaeza) [inline barcodes_generator_product/README.rst]: s/exemple/example

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The same about the icon

**Comment by lasley** [comment]
Updated, thanks @pedrobaeza

File: barcodes_generator_product/models/product_product.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014-Today GRAP (http://www.grap.coop)
+# Copyright (C) 2016-Today La Louve (http://www.lalouve.net)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models
```
Comment (Garamotte) [inline barcodes_generator_product/models/product_product.py]: `from odoo import models`

File: barcodes_generator_product/models/product_template.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014-Today GRAP (http://www.grap.coop)
+# Copyright (C) 2016-Today La Louve (http://www.lalouve.net)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, fields, api
```
Comment (Garamotte) [inline barcodes_generator_product/models/product_template.py]: `from odoo import api, fields, models`

File: barcodes_generator_product/models/product_template.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    @api.multi
+    def generate_barcode(self):
+        self.product_variant_ids.generate_barcode()
+
+    @api.onchange('barcode_rule_id')
+    def onchange_barcode_rule_id(self):
+        if self.barcode_rule_id:
+            self.generate_type = self.barcode_rule_id.generate_type
+        else:
+            self.generate_type = False
```
Comment (Garamotte) [inline barcodes_generator_product/models/product_template.py]: Is the `if` statement really useful ?
If I remember correctly, the recordset returns `False` if the relation doesn't exist.

File: barcodes_generator_product/tests/test_barcodes_generator_product.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016-Today GRAP (http://www.grap.coop)
+# Copyright (C) 2016-Today La Louve (http://www.lalouve.net)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.tests.common import TransactionCase
```
Comment (Garamotte) [inline barcodes_generator_product/tests/test_barcodes_generator_product.py]: `from odoo.tests.common import TransactionCase`

**Review by Garamotte — COMMENTED** [review]
Minor comments inline.
Otherwise, LGTM (code review only).

**Comment by legalsylvain** [comment]
ping @lasley : could you fix your code, regarding @sylvain-garancher remarks ?
it will be cool to go ahead with other PR depending on this one.
thanks !

**Comment by lasley** [comment]
Thanks for the ping @legalsylvain - another one that dropped through the cracks. Fixing now!

File: barcodes_generator_product/models/product_template.py (hunk 8 above)
Comment (lasley) [reply barcodes_generator_product/models/product_template.py]: If `self.barcode_rule_id` is `False`, `self.barcode_rule_id.generate_type` will raise an `AttributeError`. I originally tried simplifying as:
```
self.generate_type = self.barcode_rule_id and self.barcode_rule_id.generate_type or False
```
but it is too long. This was the best I could make it look:
```
self.generate_type = (
    self.barcode_rule_id and 
    self.barcode_rule_id.generate_type or
    False
)
```
But I prefer the conditional visually. 

Thoughts?

**Comment by legalsylvain** [comment]
@pedrobaeza, @sylvain-garancher : could you take a look on that updated PR ? 

kind regards.

File: barcodes_generator_product/models/product_template.py (hunk 8 above)
Comment (pedrobaeza) [reply barcodes_generator_product/models/product_template.py]: No, `self.generate_type = self.barcode_rule_id.generate_type` is not going to generate any error, as `self.barcode_rule_id` is not False, it's an empty recordset, which is not the same, returning False in any subexpression with dot notation, but not failing.

Comment (lasley) [reply barcodes_generator_product/models/product_template.py]: TIL & done - thanks @pedrobaeza & @sylvain-garancher

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash at your taste

**Comment by lasley** [comment]
Squashed - I'll merge on a 🍏  build

**Pull request merged**